Document support under Autograph for JAX array assignment (#736)
PennyLaneAI/catalyst: doc/dev/autograph.rst

...     with disable_autograph:
...         x += f()
...     return x

> Dealing with JAX array assignments inside Autograph
> ---------------------------------------------------

Reviewer: ```suggestion
In-place array assignments
--------------------------
```

Author: Done

> By default, updating a JAX array item is only possible by using the 'at' and 'set' methods:

Reviewer: ```suggestion
To update array values when using JAX, the
`JAX syntax for array assignment <https://jax.readthedocs.io/en/latest/notebooks/Common_Gotchas_in_JAX.html#array-updates-x-at-idx-set-y>`__
which uses the array ``at`` and ``set`` methods must be used:
```

Author: Done

>>> @qjit(autograph=True)
... def f(x):
...     first_dim = x.shape[0]
...     result = jnp.empty((first_dim,), dtype=x.dtype)
...
...     for i in range(first_dim):
...         result = result.at[i].set(x[i]* 2)
...
...     return result 
    

> However, Catalyst has enabled additional support for the single index array assignment syntax we all know:

Reviewer: ```suggestion
However, if updating a single index of the array, Autograph supports conversion of standard Python array assignment syntax:
```

Author: Done
